libcodechecker/analyze/store_handler.py

```diff
@@ -174,20 +174,26 @@ def is_same_event_path(report_id, events, session):
             str(ex))
 
 
-def addCheckerRun(session, storage_session, command, name, tag, username,
-                  run_history_time, version, force):
+def addCheckerRun(session_factory, storage_session, command, name, tag,
+                  username, run_history_time, version, force):
     """
     Store checker run related data to the database.
     By default updates the results if name already exists.
     Using the force flag removes existing analysis results for a run.
     """
+    session = session_factory()
+    lock_session = session_factory()
+
```

Comment:
We are using 3 sessions here? `session`, `storage_session`, `lock_session`

Reply:
Yes. Explanation is above in the pull request. To summarise:

 * `session` is the transaction that will handle the actual storage of data
 * `lock_session` is a separate transaction that handles committing the lock information in time
 * `storage_session` is the server-internal timeout handler

Reply:
Could you please add these explanations as a comment for each variable?

Reply:
An explanation exists where these extra sessions are actually used.